### ISSUE TITLE
[bug] Using this with the CORS plugin causes streaming to break

### ISSUE THREAD

**Issue by tlscipher** [elysiajs/stream#4]
When using elysia's streaming plugin with the CORS plugin, the streaming breaks. Below you can find a minimal reproducable example.

```typescript
import { Elysia } from "elysia";
import cors from "@elysiajs/cors";
import Stream from "@elysiajs/stream";
import OpenAI from "openai";

const openai = new OpenAI({
  apiKey: Bun.env.OPENAI_API_KEY,
});

const app = new Elysia()
  .use(cors())
  .get(
    '/ai',
    ({ query: { prompt } }) =>
      new Stream(
        openai.chat.completions.create({
          model: 'gpt-3.5-turbo',
          stream: true,
          messages: [{
            role: 'user',
            content: prompt
          }]
        })
      )
  )
  .listen(PORT, () => {
    console.log(`🦊 Elysia is running at on port ${PORT}`)
  });
  ```
  
  Doing this results in a content-type header of `application/json` along with a body of `{}`

**Comment by tlscipher** [comment]
Here are my depenency versions as well:
```
"@elysiajs/cors": "^0.7.1",
"elysia": "0.7.19",
"@elysiajs/stream": "^0.7.1",
"openai": "^4.12.1",
```

**Comment by bogeychan** [comment]
Hi 👋

you can do something like this as workaround:

```ts
import { Elysia } from 'elysia';
import { cors } from '@elysiajs/cors';
import { Stream } from '@elysiajs/stream';

new Elysia()
  .use(cors())
  .get('/', () => {
    const streamy = new Stream(async (stream) => {
      stream.send('hello');

      await stream.wait(1000);
      stream.send('world');

      stream.close();
    });

    return new Response(streamy.value, {
      headers: { 'Content-Type': 'text/event-stream; charset=utf-8' }
    });
  })
  .listen(8080);
```

based on:
- https://github.com/elysiajs/elysia/issues/250#issuecomment-1746961252
- https://github.com/elysiajs/elysia/blob/82f49cdff1b8c1de7ef7a1ecd06d6c97111a0f80/src/handler.ts#L393-L397

**Comment by SaltyAom** [comment]
Should be fixed by updating Elysia to 0.7.24
